спринт 18

pyclan88 opened the pull request:
- в Медиатеку добавлены TabLayout и ViewPager2 с разделами "Избранные треки" и "Плейлисты"

pyclan88 committed:
спринт 18
- в Медиатеку добавлены TabLayout и ViewPager2 с разделами "Избранные треки" и "Плейлисты"

kosstyango commented:
К сожалению, твой проект у меня не собрался: похоже, опечатка в 7 строке...
![image](https://github.com/user-attachments/assets/34b9481e-9372-4e2c-97fb-733a480e338d)

pyclan88 committed:
спринт 18
- добавлен файл MainActivity.kt с изменениями

pyclan88 committed:
спринт 18
- добавлены изменения, связанные с продолжением проигрывания после сворачивания. коммит пост ревью

kosstyango commented:
Руслан, привет!
Снимаю вопрос про перескок на следующие спринты, это я не твой проект смотрел, оказывается :(
Приношу извинения :)

kosstyango commented:
Привет, Руслан!

Выполнение требований задания:
✅ Приложение компилируется и запускается без ошибок.
✅ Приложение соответствует дизайну (допускается неполное соответствие у отображения индикатора TabLayout) в светлой и тёмной темах.
✅ Фрагменты инициализируются в адаптере при помощи методов newInstance() .
✅ Для каждого фрагмента создана ViewModel и внедрена в качестве зависимости.

Переход на экран «Медиатека».
✅ Если пользователь находится на «Главном экране», то при нажатии на кнопку «Медиатека», он переправляется на экран «Медиатека».
✅ Если приложение было переведено в фоновый режим на экране «Медиатека», то после восстановления приложения из фонового режима пользователь видит экран «Медиатека».

Переход на предыдущий экран.
✅ Нажатие на системную кнопку «Назад» (или применение аналогичного жеста) возвращает пользователя на «Главный экран».

UI экрана «Медиатека».
✅ Когда пользователь находится на экране «Медиатека», он видит заголовок «Медиатека» и кнопку «Назад».
✅ Нажатие на кнопку «Назад» аналогично нажатию на системную кнопку «Назад» и применению аналогичного жеста.

Таб-бар.
✅ Когда пользователь находится на экране «Медиатека», то он видит таб-бар, состоящий из двух табов: «Плейлисты» и «Избранные треки».

Таб «Плейлисты».
✅ Если пользователь находится на экране «Медиатека» и выбран таб «Плейлисты», то пользователь видит таб «Плейлисты» в выбранном состоянии и таб «Избранные треки» в невыбранном состоянии.

Таб «Избранные треки».
✅ Если пользователь находится на экране «Медиатека» и выбран таб «Избранные треки», то пользователь видит таб «Избранные треки» в выбранном состоянии и таб «Плейлисты» в невыбранном состоянии.

Переход в раздел «Избранные треки».
✅ Если пользователь находится на экране «Медиатека», то при нажатии на таб «Избранные треки» таб «Избранные треки» становится выбранным, а таб «Плейлисты» становится невыбранным и пользователь видит раздел «Избранные треки».
✅ Если приложение было переведено в фоновый режим на экране «Медиатека» в разделе «Избранные треки», то после восстановления приложения из фонового режима пользователь видит экран «Медиатека» с выбранным разделом «Избранные треки».

Список треков.
✅ Когда пользователь находится на экране «Медиатека», выбран раздел «Избранные треки» и у пользователя нет ни одного трека, добавленного в «Избранные треки», то пользователь видит изображение-заглушку с сообщением о том, что у него нет избранных треков.

UI раздела «Плейлисты».
✅ Когда пользователь находится на экране «Медиатека» и выбран раздел «Плейлисты», пользователь видит кнопку «Новый плейлист» и область для списка плейлистов.

Плейсхолдер (нет плейлистов).
✅ Когда пользователь находится на экране «Медиатека», выбран раздел «Плейлисты» и у пользователя ещё нет созданных плейлистов, то пользователь видит изображение-заглушку с сообщением о том, что у пользователя нет плейлистов.

Переход в раздел «Плейлисты».
✅ Если пользователь находится на экране «Медиатека», то при нажатии на таб «Плейлисты» таб «Плейлисты» становится выбранным, а таб «Избранные треки» становится невыбранным и пользователь видит раздел «Плейлисты».
✅ Если приложение было переведено в фоновый режим на экране «Медиатека» в разделе «Плейлисты», то после восстановления приложения из фонового режима пользователь видит экран «Медиатека» с выбранным разделом «Плейлисты».

⚠️ Критические замечания:
Таких нет, ты молодец! Работа принята!
 
🍏 Рекомендации, не влияющие на результат проверки работы:
Отличная работа, так держать!!!
